### ISSUE TITLE
Need any help?

### ISSUE THREAD

**Issue by gingerbeardman** [escsequence/hakchi2_4mac#1]

**Comment by foxnoodles** [comment]
I wonder if this project is dead. Pitty

**Comment by foxnoodles** [comment]
Most feature proof solution that I know of is https://github.com/libusb/libusb It's pretty straightforward and comes with nice features:

- All transfer types supported (control/bulk/interrupt/isochronous)
- 2 transfer interfaces:
- Synchronous (simple)
- Asynchronous (more complicated, but more powerful)
- Thread safe (although the asynchronous interface means that you usually won't need to thread)
- Lightweight with lean API
- Compatible with libusb-0.1 through the libusb-compat-0.1 translation layer
- Hotplug support (on some platforms). See Device hotplug event notification.

Hope this helps

**Comment by gingerbeardman** [comment]
hakchi2 is the user interface element. 

hakchi is the element that does the hard work talking to the Mini, and it's already available to be compiled for Mac. 

I thought this repo was for just the user interface part? Is that not the case?

**Comment by escsequence** [comment]
Yes it is! Wasnt 100% on the hakchi being buildable for mac. Long time ago when i checked the creator hadnt even built it for mac, since he didnt have one.

**Comment by escsequence** [comment]
Any recommendations for the ui? Was just going to replicate it 1 for 1 on the windows client of hakchi2.

**Comment by gingerbeardman** [comment]
The windows client has an awful use interface. That's how I ended up here. 

Something like Witgui (for managing Wii games on a USB drive) is a much better interface. Though I imagine something halfway between the two.

**Comment by escsequence** [comment]
Transferring this project to Qt for simplicity, going to just recreate it right now... but for the future the interface will be addressed. So just referencing the `it's already available to be compiled for Mac`. 

I have been able to build it, but it does not work.... "knock, knock" is the only message I receive.... **please let me know your secrets if you have had a successful build that works properly.** 

This will be more than just an easy "redesign", rather it will require stitching together bits and pieces and recoding various bits to get the functions working properly.

**Comment by foxnoodles** [comment]
I don't think that moving to QT was a good idea. Keeping things completely native in XCode is always better on Mac. Much more stable, better UX, less to think about layout, more native user-friendly experience overall.

**Comment by escsequence** [comment]
Gingerbeardman, my first goal before UI is to have a prototype that actually works on Mac OS. Does it build? Sure. Does Hakchi work on Mac OS? No.

**Comment by gingerbeardman** [comment]
Agreed
